src/hooks/useTreasuryPrice.tsx

```diff
@@ -0,0 +1,67 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable no-restricted-properties */
+import { PublicKey } from '@solana/web3.js'
+import { useMemo } from 'react'
+import * as tm from '@helium/treasury-management-sdk'
+import { amountAsNum } from '@helium/spl-utils'
+import { useMint } from './useMint'
+import { useTreasuryManagement } from './useTreasuryManagement'
+import { useOwnedAmount } from './useOwnedAmount'
+
+export function useTreasuryPrice(
+  fromMint: PublicKey,
+  amount: number,
+): { loading: boolean; price: number | undefined } {
+  const treasuryManagementKey = useMemo(
+    () => tm.treasuryManagementKey(fromMint)[0],
+    [fromMint],
+  )
+  const { info: treasuryAcc, loading: loadingTreasuryManagement } =
+    useTreasuryManagement(treasuryManagementKey)
+  const { info: fromMintAcc, loading: loadingFromMint } = useMint(fromMint)
+  const { info: treasuryMintAcc, loading: loadingTreasuryMint } = useMint(
+    treasuryAcc?.treasuryMint,
+  )
+
+  const {
+    amount: r,
+    decimals: rDecimals,
+    loading: loadingR,
+  } = useOwnedAmount(treasuryManagementKey, treasuryAcc?.treasuryMint)
+
+  const price = useMemo(() => {
+    if (
+      fromMintAcc &&
+      treasuryMintAcc &&
+      treasuryAcc &&
+      typeof r !== 'undefined'
+    ) {
+      // only works for basic exponential curves
+      // dR = (R / S^(1 + k)) ((S + dS)^(1 + k) - S^(1 + k))
```

Comment:
Some complicated formulas lol, are there docs we can read about this?

Reply:
Yaa haha. Let me talk to Noah and check